src/__tests__/search-and-replace-test.jsx

```diff
@@ -0,0 +1,423 @@
+var assert = require("assert");
+var StatefulEditorPage = require("../stateful-editor-page.jsx");
+var EditorPage = require("../editor-page.jsx");
+var ArticleEditor = require("../article-editor.jsx");
+var TestUtils = React.addons.TestUtils;
+
+describe("Search and Replace", function() {
+
+    describe("EditorPage", function() {
+        var editor;
+        var searchInput, replaceInput;
+        var previousButton, nextButton, replaceButton, replaceAllButton;
+
+        // Note: the example question initially was using expression widgets but
+        // this caused lots of issues because this widget type uses MathJax which
+        // isn't loaded in the testing environment.
```

Comment:
Yeah, we don't stub out mathjax (I think there's a recent mathjax-node project, but I'd rather not add that if we don't have to). If things render purely in KaTeX they're usually fine, and it's possible but tricky to do mathquill based testing for the mathquill version of the expression widget. But ultimately it's worth avoiding for tests that are focused on other things :).